drivers/flash/soc_flash_nrf_mramc.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include <string.h>
+
+ #include <zephyr/drivers/flash.h>
+ #include <zephyr/logging/log.h>
+ #include <zephyr/sys/barrier.h>
+ #include <nrfx_mramc.h>
+ #if defined(NRF_ICACHE)
+ #include <hal/nrf_cache.h>
+ #endif
+
+LOG_MODULE_REGISTER(flash_nrf_mramc, CONFIG_FLASH_LOG_LEVEL);
+
+#define DT_DRV_COMPAT nordic_mram_controller
+
+#define _ADD_SIZE(node_id) + DT_REG_SIZE(node_id)
+#define _WBS(node_id)	   DT_PROP(node_id, write_block_size),
+#define _EBS(node_id)	   DT_PROP(node_id, erase_block_size),
+
+#define MRAM_SIZE   (0 DT_INST_FOREACH_CHILD_STATUS_OKAY(0, _ADD_SIZE))
+#define ERASE_VALUE (uint8_t)NRFY_MRAMC_WORD_AFTER_ERASED
+
+/* write block size and erase block size is in the child node of
+ * mramc DT_FOREACH_CHILD_STATUS_OKAY is used to get the values
+ * from the first child node.
+ */
+static const uint32_t write_block_size_arr[] = {
+	DT_INST_FOREACH_CHILD_STATUS_OKAY(0, _WBS)
+};
+
+static const uint32_t erase_block_size_arr[] = {
+	DT_INST_FOREACH_CHILD_STATUS_OKAY(0, _EBS)
+};
+
+static const uint32_t write_block_size = write_block_size_arr[0];
+static const uint32_t erase_block_size = erase_block_size_arr[0];
+
+static int nrf_mramc_read(const struct device *dev, off_t offset, void *data, size_t len)
+{
+	ARG_UNUSED(dev);
+	uint32_t addr = NRFX_MRAMC_MAP_TO_ADDR(offset);
+
+	LOG_DBG("read: %x:%zu", addr, len);
+	nrfx_mramc_buffer_read(data, addr, len);
+	return 0;
+}
+
+static int nrf_mramc_write(const struct device *dev, off_t offset,
+			 const void *data, size_t len)
+{
+	ARG_UNUSED(dev);
+	uint32_t addr = NRFX_MRAMC_MAP_TO_ADDR(offset);
+
```

Comment:
There is expected 0 to be returned in case when len is 0 and no operation will be performed, even if other parameters are incorrect.